### PR TITLE
Bcpsdems 791 schema registry

### PR DESCRIPTION
**Pull request by leewrigh**

### CODE REVIEW

File: backend/service.edt/Kafka/ConsumerSetup.cs [hunk 1]
```diff
@@ -18,50 +22,82 @@ public static IServiceCollection AddKafkaConsumer(this IServiceCollection servic
...
-            //SslKeyLocation = config.KafkaCluster.SslKeyLocation
+            SslCertificateLocation = config.KafkaCluster.SslCertificateLocation,
+            SslKeyLocation = config.KafkaCluster.SslKeyLocation
         };
-        var producerConfig = new ProducerConfig(clientConfig)
+        producerConfig = new ProducerConfig()
         {
+            BootstrapServers = config.KafkaCluster.BootstrapServers,
             Acks = Acks.All,
+            SaslMechanism = SaslMechanism.OAuthBearer,
```
Comment (sookeke) [inline backend/service.edt/Kafka/ConsumerSetup.cs]: @leewrigh  set consumer and producer cliendId for better event tracing
ClientId = Dns.GetHostName(),

File: backend/service.edt/ServiceEvents/SchemaAwareProducer.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        var success = false;
+        var avroSerializerConfig = new AvroSerializerConfig
+        {
+            BufferBytes = 100
+        };
+
+        var schemaRegistryConfig = new SchemaRegistryConfig
+        {
+            Url = "http://pidp-kafka-apicurioregistry-pgsql.5b7aa5-dev.router-default.apps.silver.devops.gov.bc.ca/apis/ccompat/v6",
+            BasicAuthUserInfo = "registry-client-api:hPYOGgxHb8W1Wd4BAfkrvBF93IaBHI0d"
```
Comment (sookeke) [inline backend/service.edt/ServiceEvents/SchemaAwareProducer.cs]: Please remove and revoke this secret asap

**Review by sookeke — CHANGES_REQUESTED** [review]
I made some comment and please check there are some merge conflict file. 

Note there still secrets found on the pr and do not forget to revoke the secret after remediation

File: backend/service.edt/Kafka/ConsumerSetup.cs (hunk 1 above)
Comment (leewrigh) [reply backend/service.edt/Kafka/ConsumerSetup.cs]: Done

File: backend/service.edt/ServiceEvents/SchemaAwareProducer.cs (hunk 2 above)
Comment (leewrigh) [reply backend/service.edt/ServiceEvents/SchemaAwareProducer.cs]: Done

**Review by sookeke — APPROVED** [review]

**Pull request merged**